GEN-54: Move hCore to `labs` and misc fix

nonparibus opened the pull request:
## 🌟 What is the purpose of this PR?

Various small misc updates.

## 🔍 What does this change?

- Removes outdated reference to `hash-agents` app in `.prettierignore` file list
- Removes the `apps/sim-core` directory (moved in https://github.com/hashintel/labs/commit/6adcc3b63db31c2dc2409283f33eee48c3bde72e)
- Removes the `apps/sim-core-plugins/process-modeler` directory (moved in https://github.com/hashintel/labs/commit/c326a0237c0abdf8ab5dd01da7963ac1d0fcc558)

TimDiekmann reviewed (dismissed):
Do we want to keep `bug-report-hash-core.yml`?

nonparibus commented:
@TimDiekmann Good flag. Moved to `labs` in https://github.com/hashintel/labs/pull/12 and removed from here.

In anticipation of moving hEngine into `labs, also, I've migrated the issue report for it as part of this, a little bit prematurely, as well (preferable to us risking forgetting later, in my judgement).
